pkg/tracing/scopedLogger.go

```diff
@@ -2,13 +2,16 @@ package tracing
 
 import (
 	"context"
+	"runtime"
+	"strings"
 
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/trace"
 )
 
 type ScopedLogger interface {
 	StartSpan(op string, attrs ...attribute.KeyValue) (context.Context, ScopedLogger)
+	StartSpanNamed(attrs ...attribute.KeyValue) (context.Context, ScopedLogger)
```

Comment:
When should someone use StartSpan and explicitly name the op vs. using StartSpanNamed and have the op named after the current function?

Also it's not clear to me what StartSpanNamed will do. How about we tweak the name a bit so it's clear that the span will be named after the function, like `StartSpanFromFunction` or `StartSpanNamedByFunction` or something else like that?

Reply:
From my perspective, a span is for a logical unit that usually contains side affect on a state. That often happens within a function. It also helps to create a trace that easily relate back to the code. We will know exactly which function created a span. I also changed the name to `StartSpanNamedByCaller`. Let me know what do you think?

Reply:
When would someone use StartSpan vs StartSpanByCaller? Or should we remove StartSpan and never give a custom name to a span?

Reply:
I was thinking maybe someone would want multiple spans within a complex function and then they will need to manually provide names for them.